cmd/genutils/genutils.go

```diff
@@ -18,10 +18,25 @@ package genutils
 
 import (
 	"fmt"
+	"io"
 	"os"
 	"path/filepath"
 )
 
+type devNull int
+
+// NullReader is something like iotil.Discard, but a Reader
+var NullReader io.Reader = devNull(0)
+
+func (devNull) Read(p []byte) (int, error) {
+	length := len(p)
+	if length == 0 {
```

Comment:
If we were to keep this, I'd ask to please just unconditionally `return 0, io.EOF`. It doesn't make sense for callers to read a 0 byte slice, and it doesn't make sense to return N 0-bytes if they read an N-length slice..